pkg/kubectl/cmd/create_secret.go

```diff
@@ -192,3 +193,66 @@ func CreateSecretDockerRegistry(f *cmdutil.Factory, cmdOut io.Writer, cmd *cobra
 		OutputFormat:        cmdutil.GetFlagString(cmd, "output"),
 	})
 }
+
+const (
+	secretForTLSLong = `
+Create a TLS secret from the given public/private key pair.
+
+The public/private key pair must exist before hand. The public key certificate must be .PEM encoded and match the given private key.`
+
+	secretForTLSExample = `  # You can create a TLS secret using:
+  $ kubectl create secret tls tls-secret-name --cert=path to tls.cert --key=path to tls.key`
+)
+
+// NewCmdCreateSecretTLS is a macro command for creating secrets to work with Docker registries
+func NewCmdCreateSecretTLS(f *cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "tls NAME --cert=path to cert file --key=path to key file [--dry-run]",
+		Short:   "Create a TLS secret.",
+		Long:    secretForTLSLong,
+		Example: secretForTLSExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := CreateSecretTLS(f, cmdOut, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddGeneratorFlags(cmd, cmdutil.SecretForTLSV1GeneratorName)
+	cmd.Flags().String("cert", "", "Path to PEM encoded public key certificate.")
+	cmd.Flags().String("key", "", "Path to private key associated with given certificate.")
+	cmdutil.AddInclude3rdPartyFlags(cmd)
```

Comment:
Only top level kubectl should be calling that.

On Mon, Apr 25, 2016 at 12:57 PM, David Eads <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/create_secret.go
> <https://github.com/kubernetes/kubernetes/pull/24719#discussion_r60948241>
> :
>
> > +		Use:     "tls NAME --cert=path to cert file --key=path to key file [--dry-run]",
> > +		Short:   "Create a TLS secret.",
> > +		Long:    secretForTLSLong,
> > +		Example: secretForTLSExample,
> > +		Run: func(cmd *cobra.Command, args []string) {
> > +			err := CreateSecretTLS(f, cmdOut, cmd, args)
> > +			cmdutil.CheckErr(err)
> > +		},
> > +	}
> > +	cmdutil.AddApplyAnnotationFlags(cmd)
> > +	cmdutil.AddValidateFlags(cmd)
> > +	cmdutil.AddPrinterFlags(cmd)
> > +	cmdutil.AddGeneratorFlags(cmd, cmdutil.SecretForTLSV1GeneratorName)
> > +	cmd.Flags().String("cert", "", "Path to PEM encoded public key certificate.")
> > +	cmd.Flags().String("key", "", "Path to private key associated with given certificate.")
> > +	cmdutil.AddInclude3rdPartyFlags(cmd)
>
> why?
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24719/files/0980bbcac898a76e5bcde101615040924189a15c#r60948241>
>